### DISCUSSION TITLE
A removed "required field" causes error while creating a new content

### DISCUSSION THREAD

**Discussion by iedmrc** [strapi/strapi#5174] [Legacy Topics]
**Describe the bug**
I deleted a date field which was marked as required. There was no warning while removing that field but when I wanted to add new content (to that content type), I got this error message:
`error error: null value in column "video_date" violates not-null constraint`. That means either the field or the constraint still remains on db.

**Steps to reproduce the behavior**
1. Create a new content type on the` Content Type Builder` plugin.
2. Add a date (or maybe another) field and mark as required (on the Advanced Settings tab).
3. Save the content type
4. Remove the field you added.
5. Save the content type
6. Create new content under that content type
7. See the error on the console

**Expected behavior**
When removing a field on content type builder, both remove the field and constraint.

**Screenshots**

**Code snippets**
If applicable, add code samples to help explain your problem.

**System**
- Node.js version: v13.7.0
- NPM version: 6.13.6
- Strapi version: 3.0.0-beta.18.7
- Database: Postgresql 12.1
- Operating system: Macos Catalina

**Additional context**

**Comment by lauriejim** [comment]
Hello! Thank you for reporting this issue.
Here is the issue related to yours - https://github.com/strapi/strapi/issues/1114
I ping @Aurelsicoko about this issue.
This is a discussion on our side. Do we have to delete "things" in your database.

**Comment by iedmrc** [comment]
Thank you for the reply and sorry for the possible duplicate. But I think this is slightly different because it completely disables adding/creating new content because, though there is no such field anymore to present, it requires the field to be submitted.  Therefore, a new record cannot be added to that content-type (or collection) anymore. If that is the case, then strapi must forbid the user to remove a required field to avoid this caveat.

**Comment by santitone** [comment]
Hi! This is happening to me too. The only solution is to delete the collection and create it again. (Hoping that I do not add a field that needs to be deleted later)

**Comment by jonatasdiscover** [comment]
Although this is somehow related to #1114 this is not the same issue. I believe a fix to this is just about removing the constraint from the database when the field is deleted from the collection.
